samples/snippets/update_cluster_test.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import uuid
+
+from google.cloud import dataproc_v1 as dataproc
+import pytest
+
+import create_cluster
+import update_cluster
+
+
+PROJECT_ID = os.environ["GOOGLE_CLOUD_PROJECT"]
+REGION = "us-central1"
+CLUSTER_NAME = "py-cc-test-{}".format(str(uuid.uuid4()))
+NEW_NUM_INSTANCES = 5
+
+
+@pytest.fixture(autouse=True)
+def teardown():
+    yield
+
+    cluster_client = dataproc.ClusterControllerClient(
+        client_options={"api_endpoint": f"{REGION}-dataproc.googleapis.com:443"}
+    )
+    # Client library function
+    operation = cluster_client.delete_cluster(
+        request={
+            "project_id": PROJECT_ID,
+            "region": REGION,
+            "cluster_name": CLUSTER_NAME,
+        }
+    )
+    # Wait for cluster to delete
+    operation.result()
+
+
+def test_update_cluster(capsys):
+    # Wrapper function for client library function
+    create_cluster.create_cluster(PROJECT_ID, REGION, CLUSTER_NAME)
+    update_cluster.update_cluster(PROJECT_ID, REGION, CLUSTER_NAME, NEW_NUM_INSTANCES)
+
+    out, _ = capsys.readouterr()
+    assert CLUSTER_NAME in out
```

Comment:
I've been calling `dataproc.ClusterControllerClient.get_cluster(PROJECT_ID).config` to try and get the worker_config, but it doesn't let me get any deeper into the cluster object than `config`. I'm not sure how to get in there currently.

Reply:
Hmmm, the following ended up working for me when pointed to a cluster that I have with two workers. Did you also provide the region / cluster_name to `get_cluster`?

```
from google.cloud import dataproc

PROJECT_ID = "my-project"
REGION = "my-region"
CLUSTER_NAME = "my-cluster"

client = dataproc.ClusterControllerClient(
                  client_options={
                      "api_endpoint": "{}-dataproc.googleapis.com:443".format(REGION)
                   }
               )

cluster = client.get_cluster(project_id=PROJECT_ID, region=REGION, cluster_name=CLUSTER_NAME)

print(cluster.config.worker_config.num_instances) # should print an integer
 ```

Reply:
@bradmiro I haven't been able to get it working in the context of a test. My recent commit has three options I've tried: creating a fixture, creating a static cluster, and creating the cluster in the test. I still can't get in deeper than `config`. (This may be a Pythonic learning curve thing.)

Reply:
Actually does seem to be working in the current commit in CI!

Reply:
final commit ready for review!